storage/memory/log_storage.go

```diff
@@ -402,8 +412,18 @@ func (t *logTreeTX) UpdateSequencedLeaves(ctx context.Context, leaves []*trillia
 
 func (t *logTreeTX) getActiveLogIDs(ctx context.Context) ([]int64, error) {
 	var ret []int64
-	for k := range t.ts.trees {
-		ret = append(ret, k)
+	for id, tree := range t.ts.trees {
+		if tree.meta.GetDeleted() {
+			continue
+		}
+
+		switch tree.meta.TreeType {
+		case trillian.TreeType_LOG, trillian.TreeType_PREORDERED_LOG:
+			switch tree.meta.TreeState {
+			case trillian.TreeState_ACTIVE, trillian.TreeState_DRAINING:
+				ret = append(ret, id)
+			}
+		}
```

Comment:
Can we commonize this as a method on `TreeStorage` ?

Reply:
I just made it a free function, rather than putting it on `TreeStorage`, since it only need the map of tree IDs to trees.